tests/cli/test.cli.js

```diff
@@ -0,0 +1,41 @@
+import {expect} from 'chai';
+import {spy} from 'sinon';
+
+import {Program} from '../../src/cli';
+
+
+describe('cli.Program', () => {
+
+  it('executes a command callback', () => {
+    let thing = spy(() => new Promise((resolve) => resolve()));
+    let program = new Program(['thing'])
+      .command('thing', 'does a thing', thing);
+    return program.run()
+      .then(() => {
+        expect(thing.called).to.be.equal(true);
```

Comment:
I don't have strong feelings about it. I like the readability of expect but it is slightly harder to type than `assert.equal()` so meh.